src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -342,11 +341,8 @@ public function nack(\AMQPEnvelope $message, string $queueName, int $flags = AMQ
 
     public function setup(): void
     {
-        if (!$this->channel()->isConnected()) {
-            $this->clear();
```

Comment:
this should also happen when auto_setup is not enabled to reconnect when necessary. so I moved it outside setup()